main.go

```diff
@@ -85,13 +87,24 @@ func registerAgent(ctx context.Context) {
 		if client, err := agentendpoint.NewClient(ctx); err != nil {
 			logger.Errorf(err.Error())
 		} else if err := client.RegisterAgent(ctx); err != nil {
-			logger.Errorf(err.Error())
+			if s, ok := status.FromError(err); ok {
+				if s.Code() == codes.FailedPrecondition {
+					logger.Infof("Agent rejected.")
+					break
+				}
+			} else {
+				logger.Errorf(err.Error())
+			}
 		} else {
 			// RegisterAgent completed successfully.
 			return
 		}
 		time.Sleep(5 * time.Minute)
 	}
+	// Agent rejected, just wait for the possible parent context termination.
+	select {
+	case <-ctx.Done():
```

Comment:
We can't hang forever, unless we know that this error will never ever change. We should instead just keep it in the loop and keep retrying (but only log the first time.)